core/shared/src/main/scala/zio/ZTraceElement.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2021 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio
+
+import zio.internal.stacktracer.Tracer
+import zio.stacktracer.TracingImplicits.disableAutoTrace
+
+object ZTraceElement {
+
+  def apply(location: String, file: String, line: Int): ZTraceElement =
+    Tracer.instance(location, file, line)
+
+  val empty: ZTraceElement =
+    Tracer.instance.empty
+
+  def fromJava(stackTraceElement: StackTraceElement): ZTraceElement =
+    ZTraceElement(
+      stackTraceElement.getClassName + "." + stackTraceElement.getMethodName,
+      stackTraceElement.getFileName,
+      stackTraceElement.getLineNumber
+    )
+
+  def toJava(trace: ZTraceElement): Option[StackTraceElement] =
+    trace match {
+      case ZTraceElement(location, file, line) =>
+        val last = location.lastIndexOf(".")
+
+        val (before, after) = if (last < 0) ("", "." + location) else location.splitAt(last)
+
+        def stripSlash(file: String): String = {
```

Comment:
I am not sure if this is still necessary, because if we are only capturing the file name, then there won't be any path separators anymore (previously, this converted, e.g. `/Users/John/.../ZIO.scala` into `ZIO.scala`).